.github/workflows/test.yml

```diff
@@ -5,7 +5,20 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: sudo apt-get install -y valgrind && sudo apt-get install -y cppcheck
+      - name: CPPCheck Static Code Analysis
+        run: cppcheck --enable=all ./*.cpp 2> cppcheck_report.txt && if grep -q "(warning)" "./cppcheck_report.txt"; then exit 1; fi && if grep -q "(error)" "./cppcheck_report.txt"; then exit 1; fi
+      - name: Upload CPPCheck report as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: cppcheck_report
+          path: cppcheck_report.txt
       - name: Build
-        run: g++ -std=c++17 *.cpp -o output
-      - name: Test
+        run: g++ -Wall -Werror -Wno-reorder -std=c++17 *.cpp -o output
+      - name: Test run after build
+        run: ./output
+      - name: Valgrind memory check
+        run: valgrind ./output
+      - name: In-Out tests
```

Comment:
Az in-out teszteknél úgyis le van tesztelve a futás, ezért a "Test run after build" lépés felesleges. A memchecknél viszont ha van leak, akkor ne kapjon pipát a workflow.